robot_smach_states/benchmark/benchmark_grasping.py

```diff
@@ -0,0 +1,224 @@
+    single = subparsers.add_parser(name='single', description="Measure the time it takes to grasp an entity when starting from a given pose. "
+                                                 "After the grasping is done, go back to the waypoint, "
+                                                 "turn around and drop the item, "
+                                                 "then turn around once more to be in the start configuration again")
+    single.add_argument("cls", type=str,
+                        help="class of entity to grasp, eg. 'coke' or one of {} if you don't care".format(ANY_OPTIONS))
+    single.add_argument("support", type=str,
+                        help="ID of entity to grasp FROM ('grasp' entity is on-top-of this 'support' entity), eg. 'cabinet'")
+    single.add_argument("waypoint", type=str,
+                        help="ID of waypoint for start and end position of the robot")
+    single.add_argument("--inspect_from_area", type=str,
+                        help="What area of the support-entity to inspect from", default='in_front_of')
+
+    batch_config_fields = ['cls', 'support', 'waypoint', 'inspect_from_area', 'search_area']
+    batch = subparsers.add_parser(name='batch', description="Perform the single case repeatedly, "
+                                                            "taking configurations from a .csv file, "
+                                                            "with columns {}".format(','.join(batch_config_fields)))
+    batch.add_argument("--configuration", type=str, default='grasp_benchmark_config.csv')
+
+    args = parser.parse_args()
+    rospy.init_node("benchmark_grasping")
+
+    with open(args.output, 'a+') as csv_file:
+        reader = csv.DictReader(csv_file)
+
+        fields = ['timestamp', 'robot',
+                  'start_waypoint',
+                  'expected_class', 'observed_class',
+                  'id',
+                  'inspect_duration',
+                  'grab_duration',
+                  'x', 'y', 'z']
+        results_writer = csv.DictWriter(csv_file, fieldnames=fields, extrasaction='ignore')  # Ignore extra keys in the dicts
+
+        if reader.fieldnames != results_writer.fieldnames:
+            results_writer.writeheader()
+
+        robot = get_robot(args.robot)
+
+        if args.subcommand == 'single':
+            single_item(robot, results_writer,
+                        args.cls, args.support, args.waypoint, args.inspect_from_area,
+                        args.non_strict_class)
+        elif args.subcommand == 'batch':
+            with open(args.configuration) as config_file:
+                config_reader = csv.DictReader(config_file)
+                assert config_reader.fieldnames == batch_config_fields, "CSV file need fields {}".format(','.join(batch_config_fields))
+
+                configs = []
+
+                environment_config_description = ""
```

Comment:
What is the use of this? It is only used in line 212 but as far as I see it there is no use...

Reply:
Maybe the overall description can be removed